index.html

```diff
@@ -86,25 +86,311 @@ <h3>Use-case #4: Avoiding "Hall of Mirrors"</h3>
-      <h2>The Capture-Handle Mechanism</h2>
+    <section id="capture-handle-identification-actions">
+      <h2>The Capture-Handle Actions Mechanism</h2>
       <p>
-        The capture-handle mechanism consists of two main parts - one on the captured side, one on
+        The capture-handle actions mechanism consists of two parts - one on the captured side, one on
         the capturing side.
       </p>
       <ul>
         <li>
-          <a href="#captured-side">Captured applications</a> opt-in to exposing information by
+          <a href="#captured-side-actions">Captured applications</a> opt-in by registering support for
+          standard actions they handle by calling {{MediaDevices/setSupportedCaptureActions}}.
+        </li>
+        <li>
+          <a href="#capturing-side-actions">Capturing applications</a> may trigger these actions using
+          {{MediaStreamTrack/sendCaptureAction}}.
+        </li>
+      </ul>
+      <div class="note">
+        There is disagreement on whether actions should be specified here or in a separate document.
+      </div>
+    </section>
+    <section id="captured-side-actions">
+      <h2>Captured Side for Actions</h2>
+      <p>
+        Applications in top-level documents can declare the [=capture actions=]
+        they support, if any. They would typically do so before even knowing if
+        they are being captured. These actions enable the user to control the
+        progression of the captured session without leaving the capturer document.
+        Supported actions are declared by calling {{MediaDevices/setSupportedCaptureActions}}
+        with an array of the names of actions the application is prepared to respond to.
+      </p>
+      <section id="set-supported-capture-actions">
+        <h3>Registering and responding to capture actions</h3>
+        <p>
+          {{MediaDevices}} is extended with a method - {{MediaDevices/setSupportedCaptureActions}} -
+          which accepts an array of {{DOMString}}s. By calling this method, an application
+          registers with the user agent a set of zero or more [=capture actions=] it wishes to
+          respond to.
+        </p>
+        <p>
+          <dfn>Capture actions</dfn> are values defined in {{CaptureAction}}.
+          They are meant to be interpreted as instructions from the user (through the user agent)
+          to control the advancement of the presentation of the captured session, however the
+          captured application wishes to define this. Actions are designed to originate from the
+          user through interactive controls provided by the capturer document, where they
+          [=consume user activation=].
+        </p>
+        <pre class="idl">
+          partial interface MediaDevices {
+            undefined setSupportedCaptureActions(sequence&lt;DOMString&gt; actions);
+            attribute EventHandler oncaptureaction;
+          };
+
+          enum CaptureAction {
```

Comment:
This needs to be an extendible set; IDL enums are not well suited for extensibility.

I suggest DOMString, just as we do for things like codec types and SVC structures.

Reply:
I'll fix CaptureActionEventInit to use DOMString based on this feedback. Probably shouldn't use an enum there.